File: frontend/src/types/quota.ts
export interface QuotaWindowUsage {
  used: number;
  reserved: number;
  limit: number;
  remaining: number;
  utilization_percent: number;
  resets_at: string;
}

export interface ModelQuotaUsage {
  model: string;
  requests_minute: QuotaWindowUsage;
  requests_day: QuotaWindowUsage;
  tokens_minute: QuotaWindowUsage;
  tokens_day: QuotaWindowUsage;
}

export interface QuotaUsageResponse {
  models: ModelQuotaUsage[];
}

File: backend/app/schemas/quota.py
from datetime import datetime

from pydantic import BaseModel


class QuotaWindowUsage(BaseModel):
    used: int
    reserved: int
    limit: int
    remaining: int
    utilization_percent: float
    resets_at: datetime


class ModelQuotaUsage(BaseModel):
    model: str
    requests_minute: QuotaWindowUsage
    requests_day: QuotaWindowUsage
    tokens_minute: QuotaWindowUsage
    tokens_day: QuotaWindowUsage


class QuotaUsageResponse(BaseModel):
    models: list[ModelQuotaUsage]
